lib/3scale/backend/listener.rb

```diff
@@ -525,8 +525,15 @@ def check_transactions_validity(transactions)
       # "response" and "request". Those fields are not used anymore, but some
       # callers are still sending them. We want to filter them to avoid storing
       # them in the job queues, decoding them, etc. unnecessarily.
-      def log_without_unused_attrs(log)
-        log.select! { |k| k == 'code' }
+      # Also, discard logs that don't have the 'code' attribute
+      def filter_log_param(params, key = :log)
+        log_param = params[key]
+
+        if log_param.is_a?(Hash) && log_param['code']
+          params[key] = log_param.slice('code')
+        else
+          params.delete(key)
```

Comment:
Good detective work. Shouldn't we retunr 4xx in case of bad parameters though? Instead of silently ignoring the input?